core/src/test/java/edu/wpi/grip/core/http/GripServerTest.java

```diff
@@ -0,0 +1,216 @@
+
+    @Test
+    public void testSuccessfulPostHandler() throws IOException {
+        final String path = "/testAddPostHandler";
+        final byte[] testBytes = "testAddPostHandler".getBytes();
+        final boolean[] didHandle = {false};
+        PostHandler handler = bytes -> {
+            didHandle[0] = true;
+            return Arrays.equals(bytes, testBytes);
+        };
+
+        instance.addPostHandler(path, handler);
+        HttpResponse response = doPost(path, testBytes);
+        assertEquals("The server should return an OK status (200)", response.getStatusLine().getStatusCode(), 200);
+        assertTrue("Handler should have run", didHandle[0]);
+        assertTrue("The server should have removed the handler", instance.removePostHandler(handler));
+    }
+
+    @Test
+    public void testUnsuccessfulPostHandler() throws IOException {
+        final String path = "/testAddPostHandler";
+        final byte[] testBytes = new byte[0];
+        final boolean[] didHandle = {false};
+        PostHandler handler = bytes -> {
+            didHandle[0] = true;
+            return false;
+        };
+
+        instance.addPostHandler(path, handler);
+        HttpResponse response = doPost(path, testBytes);
+        assertEquals("Server should return an internal error (500)", response.getStatusLine().getStatusCode(), 500);
+        assertTrue("Handler should have run", didHandle[0]);
+    }
+
+    @Test
+    public void testAddRemoveDataSupplier() {
+        String name = "testDataSuppliers";
+        Supplier<?> supplier = () -> name;
+        instance.addDataSupplier(name, supplier);
+        assertTrue("Server should have a data supplier for " + name, instance.hasDataSupplier(name));
+        instance.removeDataSupplier(name);
+        assertFalse("Server should no longer have a data supplier for " + name, instance.hasDataSupplier(name));
+    }
+
+    @Test
+    public void testNoSupplier() throws IOException {
+        assertEquals("There shouldn't be any data on the server", "{}", doGet(GripServer.DATA_PATH));
+        assertEquals("There shouldn't be any data on the server", "{}", doGet(GripServer.DATA_PATH + "?no_data_here"));
+    }
+
+    @Test
+    public void testWithSupplier() throws IOException {
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        final String dataName = "testWithSupplier";
+        final Map data = ImmutableMap.of("foo", "bar");
+        instance.addDataSupplier(dataName, () -> data);
+        assertEquals("Generated json was malformed", gson.toJson(ImmutableMap.of(dataName, data)), doGet(GripServer.DATA_PATH + "?" + dataName));
+        assertEquals("Generated json was malformed", gson.toJson(ImmutableMap.of(dataName, data)), doGet(GripServer.DATA_PATH));
+        assertEquals("There shouldn't be any data on this path", "{}", doGet(GripServer.DATA_PATH + "?no_data_here"));
+    }
+
+    @Test
+    public void testStartStop() throws GripException, IllegalStateException {
```

Comment:
This will thrown an exception if something fails, no need to put in an `assert` or `fail` condition